roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -38,6 +41,41 @@ impl JobDeclaratorDownstream {
     }
 }
 
+pub fn clear_old_mempool_transactions(
+    old_declare_mining_job: Option<DeclareMiningJob>,
+    mempool: Arc<Mutex<JDsMempool>>,
+) -> Result<(), Error> {
+    // If there is an old declared mining job, remove its transactions from the mempool
+    if let Some(old_job) = old_declare_mining_job {
+        // Retrieve necessary data from the old job
+        let old_tx_hash_list = old_job.tx_short_hash_list.inner_as_ref();
+        let old_nonce = old_job.tx_short_hash_nonce;
+
+        // Get the mempool of the old transaction IDs
+        let old_short_id_mempool = mempool
+            .safe_lock(|x| x.to_short_ids(old_nonce))
+            .unwrap()
+            .unwrap();
```

Comment:
Why do you convert mempool into short_ids? Isn't it already made by Txid?

Reply:
I have modified that in the new commit i will push now. Thanks @GitGab19

Reply:
@GitGab19 Here